Update kata to kata-containers 3.29.0

stalb opened the pull request:
###  Update kata to kata-containers 3.15.0

This PR updates the kata addon #256, replacing usage of snap kata-container with deployment of kata-deploy helm release which support microk8s since kata-containers 3.15.0

*Also verify you have:*
* [x] Read the [contributions](https://github.com/ubuntu/microk8s/blob/master/CONTRIBUTING.md) page.
* [x] Submitted the [CLA form](https://ubuntu.com/legal/contributors/agreement), if you are a first time contributor.

stalb committed:
Update kata to 3.15.0

louiseschmidtgen commented on addons/kata/enable:
```diff
@@ -1,118 +1,97 @@
+KATA_DEPLOY_URL=https://github.com/kata-containers/kata-containers/releases/download/${VERSION}/kata-deploy-${VERSION}.tgz
+HELM_OPTS=
+
+# Enable dependencies
+MICROK8S_ENABLE="${SNAP}/microk8s-enable.wrapper"
+"${MICROK8S_ENABLE}" helm3
```
```suggestion
HELM="$SNAP/microk8s-helm3.wrapper"
```
Let's rather fail if helm is not enabled, by also adding a prereqs check:
```
do_prerequisites() {
  "$SNAP/microk8s-enable.wrapper" helm3
}
  ```

stalb replied:
Suggestion applied

louiseschmidtgen commented on addons/kata/enable:
```diff
@@ -1,118 +1,97 @@
+
+. "${SNAP}/actions/common/utils.sh"
+
+debug="false"
+shims="clh cloud-hypervisor dragonball fc qemu qemu-coco-dev qemu-runtime-rs qemu-sev qemu-snp qemu-tdx stratovirt qemu-nvidia-gpu qemu-nvidia-gpu-snp qemu-nvidia-gpu-tdx"
+defaultShim="qemu"
+createRuntimeClasses="true"
+createDefaultRuntimeClass="true"
+k8sDistribution="microk8s"
```
Could we please use these as the default install options and pass a helm values file similar to the implementation for the falco addon: https://github.com/canonical/microk8s-community-addons/blob/main/addons/falco/enable#L25?

stalb replied:
Suggestion applied

louiseschmidtgen commented on addons/kata/enable:
```diff
@@ -1,118 +1,97 @@
+createRuntimeClasses="true"
+createDefaultRuntimeClass="true"
+k8sDistribution="microk8s"
+VERSION="3.15.0"
+NAMESPACE="kata-containers"
```
```suggestion
KATA_VERSION="3.15.0"
KATA_NAMESPACE="kata-containers"
```

stalb replied:
Suggestion applied

louiseschmidtgen reviewed (commented):
Thank you so much for your work on the kata addon @stalb! 
I've left some comments, please reach out to me if something is unclear.
We appreciate your help.

stalb committed:
Apply suggestions

ktsakalozos commented on addons.yaml:
```diff
@@ -124,11 +124,14 @@ microk8s-addons:
       supported_architectures:
+        - arm64
         - amd64
+        - s390x
```
Have you tested kata on all these new architectures?

stalb replied:
Personally no, I'm pretty limited in terms of servers I can use for experimentation, however kata-container lists it in their [supported platforms] (https://github.com/kata-containers/kata-containers?tab=readme-ov-file#platform-support).

I therefore assume that kata-containers can run on s390x (they do tests for s390x...) and I hope that _microk8s-community-addons_ test might also confirm that it can be deployed on s390s with this plugin and microk8s...

louiseschmidtgen reviewed (commented):
Hi @stalb,

apologies for the delay, could you please fill out the [CLA form](https://ubuntu.com/legal/contributors/agreement) again?

Thank you for your contribution!

stalb committed:
Add compatibility with next kata-containers version (3.25)

louiseschmidtgen commented:
Closing this PR — the target version (3.15.0) is now significantly behind upstream (3.29.0), and the CLA may have expired. The approach is sound (switching to kata-deploy Helm chart), so if this is revisited, a fresh PR targeting a current release would be ideal. Thank you for the work @stalb!

stalb commented:
@louiseschmidtgen, Last year I asked you, to review this PR before the expiration of my CLA., but you didn't have time for that. So I had to to renew it. 

Now you're closing this PR because it's too old. 

The current PR allows you to choose the version of the chart and is able to install recent versions of kata-containers.
To change the default installed version, is also a one lime modification of the plugin.

Meanwhile, Canonical continues to promote the old kata plugin on its microk8s website. So far so good.

louiseschmidtgen committed:
fix: update kata-containers to 3.29.0, restrict to tested amd64 arch

- Bump default KATA_VERSION from 3.15.0 to 3.29.0
- Restrict supported_architectures to amd64 only (arm64/s390x/ppc64le
  untested; arm64 also removed from kata-deploy upstream in 3.29.0)
- Fix disable script: use microk8s-helm3.wrapper (consistent with enable)
- Fix usage string: ${VERSION} -> ${KATA_VERSION}
